test/utils/browser.js

```diff
@@ -26,36 +48,88 @@ exports.openMusic = function (options) {
+    }
     global.browser = this.browser;
   });
   before(function openBrowser (done) {
-    this.browser.init({browserName: 'chrome'}, done);
+    var that = this;
+    // DEV: Force the Windows 8 platform as it appears to run our test more
+    //      stable than other platforms
+    this.browser.init({
+      browserName: 'chrome',
+      name: 'gmusic.js - ' + testName,
+      project: 'gmusic.js Selenium tests',
+      platform: 'WIN8',
+      'browserstack.user': BROWSERSTACK_USER,
+      'browserstack.key': BROWSERSTACK_KEY
+    }, function setBrowserVariables () {
+      // DEV: Maximize the browser window to minimize the possiblity of "element not visible erros"
+      that.browser.maximize();
+      that.browser.setAsyncScriptTimeout(ASYNC_SCRIPT_TIMEOUT, done);
+    });
   });
   before(function navigateToMusicBeforeLogin (done) {
     this.browser.get(url, done);
   });
-  before(function handleLoginViaCookies (done) {
+  before(function handleLogin (done) {
     var browser = this.browser;
-    async.forEach(cookies, function setCookies (cookie, cb) {
-      // If the cookie is not for .google.com, skip it
-      // DEV: As discovered by Burp suite's repeater, we only need `SID`, `HSID`, `SSID` but this is simpler
-      if (cookie.domain !== '.google.com') {
-        process.nextTick(cb);
-      // Otherwise, set it
-      } else {
-        browser.setCookie(cookie, cb);
+    var that = this;
+
+    async.waterfall([
+      function findSignInButton (cb) {
+        browser.waitForElementByCssSelector('[data-action=signin]', asserters.isDisplayed, cb);
+      },
+      function clickSignInButton (el, cb) {
+        el.click(cb);
+      },
+      function findEmailInput (cb) {
+        browser.waitForElementById('Email', asserters.isDisplayed, cb);
+      },
+      function enterEmailIntoInput (el, cb) {
+        exports.simpleExecute.call(that, function setEmailInputValue () {
+          document.getElementById('Email').value = arguments[0];
+        }, [GOOGLE_MUSIC_JS_EMAIL], cb);
+      },
+      function findNextButton (executeResult, cb) {
+        browser.waitForElementById('next', asserters.isDisplayed, cb);
+      },
+      function clickNextButton (el, cb) {
+        el.click(cb);
+      },
+      function findPasswordInput (cb) {
+        browser.waitForElementById('Passwd', asserters.isDisplayed, cb);
+      },
+      function enterPasswordIntoInput (el, cb) {
+        exports.simpleExecute.call(that, function setPasswordInputValue () {
+          document.getElementById('Passwd').value = arguments[0];
+        }, [GOOGLE_MUSIC_JS_PASSWORD], cb);
+      },
+      function findLoginButton (executeResult, cb) {
+        browser.waitForElementById('signIn', asserters.isDisplayed, cb);
+      },
+      function clickLoginButton (el, cb) {
+        exports.simpleExecute.call(that, function clickSignInButton () {
+          document.getElementById('signIn').click();
+        }, [], cb);
```

Comment:
I was about to open up the `_execute` PR when I realized that we shouldn't be using `execute` for this logic at all. Is there any reason we aren't using `type` and `click` for the elements?

`POST /session/:sessionId/element/:id/value`

https://github.com/admc/wd/blob/v0.4.0/doc/api.md

```js
function enterPasswordIntoInput (el, cb) {
  el.type(GOOGLE_MUSIC_JS_PASSWORD, cb);
}
// ...
function clickLoginButton (el, cb) {
  el.click(cb);
}
```